src/Passes/Source/Generators/LlvmPassesConfiguration.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class LlvmPassesConfiguration
+    {
+      public:
+        // Default constructor which sets the standard pipeline.
+        LlvmPassesConfiguration();
+
+        // Setup and pre-fabricated configurations
+
+        /// Setup function that registers the different LLVM passes available via LLVM component.
+        void setup(ConfigurationManager& config);
+
+        /// Static function creates a new configuration where all transformations/validation requirements
+        /// are disabled.
+        static LlvmPassesConfiguration createDisabled();
+
+        // Configuration interpretation
+        //
+
+        /// Returns true if the configuration disables all effects of this component. The effect of this
+        /// function being true is that registered component should have no effect on transformation
+        /// and/or validation of the QIR.
+        bool isDisabled() const;
+
+        /// Returns true if the current configuration is equivalent to the default configuration.
+        bool isDefault() const;
```

Comment:
I have an impression that all the `isDefault()` functionality is only used in tests and nowhere else (a functionality for the sake of testing only). If that's the case then why not remove all the `isDefault()`?
So far I have never met any need in a functionality like this.

Reply:
Removed.